### PR TITLE
bring popup editor to canjs and enable preview

### PR DESCRIPTION
**Pull request by janebitovi**
for CCALI/a2jauthor#306

paired with https://github.com/CCALI/a2jviewer/pull/228

### CODE REVIEW

**Comment by janebitovi** [comment]
Also fixing: interview upload refresh list and scroll to new guide; do the same after completing a merge-tool guide

**Comment by janebitovi** [comment]
also fixing: unassigned variable warning when you press preview (and make it more specific)

**Comment by janebitovi** [comment]
also fixing: file upload page fields had legacy behavior that allow the value to be any string at all if it ends with a valid file name after a forward or backward slash

File: app.stache [hunk 1]
```diff
@@ -196,9 +196,9 @@
           <a on:click="mergeTool = false" class="btn btn-default" style="position: absolute; right: 2rem; margin-top: 3.5rem;">Exit without saving</a>
           <merge-tool vm:targets:from="interviews" vm:sources:from="interviews" reloadInterviews:from="reloadInterviews" open:bind="mergeTool" />
         {{/if}}
-        <interviews-toolbar interviews:from="interviews"></interviews-toolbar>
+        <interviews-toolbar interviews:bind="interviews"></interviews-toolbar>
```
Comment (mitch-mitchel) [inline app.stache:199]: this used to be `bind` in the past, and i can't recall if the change to `from/to` was solving something a problem or warning, or if i was just 'optimizing'. worth keeping an eye on is all.

File: src/interviews/interviews.js [hunk 2]
```diff
@@ -19,14 +19,18 @@ export const InterviewsVM = DefineMap.extend('InterviewsVM', {
...
 
   interviews: {
-    value ({ lastSet, listenTo, resolve }) {
-      this.interviewsPromise.then((interviews) => {
-        resolve(interviews)
-      })
-
-      listenTo(lastSet, (interviews) => {
-        resolve(interviews)
+    value (prop) {
```
Comment (mitch-mitchel) [inline src/interviews/interviews.js:22]: my above comment might actually be solved with your prop/resolver pattern. i feel like it was likely a canjs warning about child updating parent.

File: src/merge-tool/merge-tool.js [hunk 3]
```diff
@@ -333,7 +333,7 @@ export const MergeToolVM = DefineMap.extend('MergeToolVM', {
             fileList,
             templateIds
           }, function (data) {
-            console.log(newgid, data)
+            // console.log(newgid, data)
```
Comment (mitch-mitchel) [inline src/merge-tool/merge-tool.js:336]: i think i left this in there ... my bad.

File: src/pages-tab/components/file-picker/file-picker.js [hunk 4]
```diff
@@ -32,12 +32,24 @@ export const FilePicker = DefineMap.extend('FilePicker', {
   items: {},
   accept: { type: 'text', default: '' },
 
+  legacyFlattenPath (file) {
```
Comment (mitch-mitchel) [inline src/pages-tab/components/file-picker/file-picker.js:35]: i like this pattern of naming the old ways with 'legacy' - easy way to search this stuff out in the future, or know why it's 'not canjs-i-fied' at a glance

**Review by mitch-mitchel — APPROVED** [review]
minor notes only

File: app.stache (hunk 1 above)
Comment (janebitovi) [reply app.stache:199]: It used to be "to", I changed it to "from" at the start of the merge tool work not realizing the upload button was relying on it. I changed the 'value' behavior in the other component to allow bind to work on both spots so everything is happy

File: src/interviews/interviews.js (hunk 2 above)
Comment (janebitovi) [reply src/interviews/interviews.js:22]: yep, this keeps it all happy :)

File: src/pages-tab/components/file-picker/file-picker.js (hunk 4 above)
Comment (janebitovi) [reply src/pages-tab/components/file-picker/file-picker.js:35]: "You allow the saved value to be literally anything as long as it ends with a slash and a valid file name after it?"
"Legacy"
"oh."
:shipit: 
lol

**Pull request merged**